tests/views/ApiViewTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @covers ApiView
+ */
+class ApiViewTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testThatAReturnCodeSetViaHeaderIsReturnedWhenNoOtherReturnCodeIsSet()
+    {
+        $view = new ApiView();
+
+        header('Foo: bar', false, 201);
+
+        ob_start();
+        $view->render('');
+        $this->assertEquals('', ob_get_contents());
+        ob_end_clean();
+
+        $this->assertEquals(201, http_response_code());
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testThatAReturnCodeSetViaSetHeaderIsReturnedEvenThoughAHeaderCodeIsSet()
+    {
+        $view = new ApiView();
+
+        $view->setResponseCode(202);
+        header('Foo: bar', false, 201);
+
+        ob_start();
+        $view->render('');
+        $this->assertEquals('', ob_get_contents());
+        ob_end_clean();
+
+        $this->assertEquals(202, http_response_code());
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testThatHeadersAreSet()
+    {
+        $view = new ApiView();
+
+        $view->setResponseCode(202);
+        header('Foo: bar', false, 201);
+
+        ob_start();

```

Comment:
Sadly not. As **this** tests exactly that part that outputs the view. And the output is done via ```echo``` so I need to get the output buffer here. This test only checks whether the correct header is returned as that was changed (and not tested before) in the refactoring.

A separate test that checks the correct string value is returned should be added with a separate PR.